otterai/llm_client.py

```diff
@@ -0,0 +1,21 @@
+import os
+import httpx
+from langchain_openai import ChatOpenAI
+
+class LLMClient:
+    _instance = None
+
+    def __new__(cls) -> 'LLMClient':
+        if cls._instance is None:
+            cls._instance = super(LLMClient, cls).__new__(cls)
+            cls._instance.client = ChatOpenAI(
+                model_name=os.getenv('INPUT_MODEL', 'gpt-4-turbo-preview'),
+                http_async_client=httpx.AsyncClient(timeout=60.0),
+                api_key=os.getenv('INPUT_OPENAI_API_KEY'),
```

Comment:
🔒 The 'api_key' is retrieved from an environment variable; consider adding a check to ensure it's not empty or None.